consensus/src/messages/handlers.rs

```diff
@@ -456,6 +502,38 @@ impl<N: Network> Handle<N, ResponseTransactionsProof, Arc<RwLock<Blockchain>>>
         }
 
         ResponseTransactionsProof { proof, block }
+    } else {
+        // If we couldn't find the transaction in our history store, then we cannot prove it.
+        ResponseTransactionsProof {
+            proof: None,
+            block: None,
+        }
+    }
+}
+
+#[cfg(feature = "full")]
+impl<N: Network> Handle<N, ResponseTransactionsProof, Arc<RwLock<Blockchain>>>
+    for RequestTransactionsProof
+{
+    fn handle(
+        &self,
+        _peer_id: N::PeerId,
+        blockchain: &Arc<RwLock<Blockchain>>,
+    ) -> ResponseTransactionsProof {
+        if self.hashes.is_empty() {
+            // If we are not given a list of transactions then there is nothing to do
+            return ResponseTransactionsProof {
+                proof: None,
+                block: None,
+            };
+        }
+
+        // Handle the different cases: if we are provided a block number (to generate the proof) or not
+        if let Some(block_number) = self.block_number {
+            prove_txns_with_block_number(blockchain, &self.hashes, block_number)
+        } else {
+            prove_transaction(blockchain, &self.hashes[0])
```

Comment:
So this only works for requesting one transaction, not for multiple without block. Is that on purpose? Could it also prove multiple hashes, or is that too much of a can of worms to open?

Reply:
Because if we support multiple txns, then we would need to group them by finalized epoch/current epoch/current batch and potentially it would require returning multiple blocks, etc..